src/snapshot.rs

```diff
@@ -36,28 +55,36 @@ impl SnapshotSource for Snap {
 }
 
 // Ledger file format is the default serde JSON representation of VecM<(LedgerKey, LedgerEntry)>
-pub fn read(input_file: &std::path::PathBuf) -> Result<OrdMap<LedgerKey, LedgerEntry>, Error> {
-    let mut res = OrdMap::new();
+pub fn read(
+    input_file: &std::path::PathBuf,
+) -> Result<(LedgerInfo, OrdMap<LedgerKey, LedgerEntry>), Error> {
```

Comment:
The `read` function could return `SerializableState`. I don't feel strongly about this, but it might make the calling code a little neater.

Reply:
My comment above applies here, as well as the fact that this returns an `OrdMap` of ledger entries, while `SerializableState` contains a `VecM`.